### PR TITLE
Support for Litex VexRiscV

### PR DESCRIPTION
**Pull request by aenrbes**
Add initial support for Litex with softcore CPU VexRiscV SoC [LiteX's website](https://github.com/enjoy-digital/litex), device drivers such as GPIO/I2C/SPI are missing.

Note that Litex vexriscv has a rom bootloader integrated in fpga gateware, it load image from tftp/flash to on board ddr3. 

The ostest passed. Basic nsh commands such as uname/ps/free/sleep work fine.Tested on Digilent ARTY_A7 board.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,5 @@ core
 /.project
 /tags
 /TAGS
+.vscode/settings.json
+.vscode/c_cpp_properties.json
```
Comment (Ouss4) [inline .gitignore]: I don't think that editor specific files belong to the top level .gitignore.
You can ignore them locally.
@patacongo what's your take?

**Comment by davids5** [comment]
@aenrbes Nice! 

Since this is a new contribution (all new files) the licences needs to be the Apache one (@Ouss4 is there a template or instructions?) and an the contributor will need to submit an ICLA https://www.apache.org/licenses/icla.pdf

**Comment by Ouss4** [comment]
A template for the Apache license? 
Here is one https://github.com/apache/incubator-nuttx/blob/master/sched/sched/sched_getcpu.c
The LICENSE file also has [an example.](https://github.com/apache/incubator-nuttx/blob/master/LICENSE#L191)

**Comment by aenrbes** [comment]
@Ouss4 Thank you, does this means I need to change all submitted files like that templates does？

**Comment by davids5** [comment]
@Ouss4 do we have a Github PR/Issue template (and access to it) we can add all this info to?

**Comment by Ouss4** [comment]
@aenrbs New files need to have an Apache license. Kconfig, README and auto generated files like defconfig don't need a license.
Note that the license holder in all of this PRs files is Gregory Nutt. Gregory Nutt has already submitted an ICLA and gave his approval to change his license header.

@davids5  As of now, we don't. Some has adopted their own templates when submitting PRs.
We may want to add Github's templates for PRs/Issues. Maybe we bring this to the dev list for a discussion on what should be there?

**Comment by patacongo** [comment]
The required form, the "teplate", is in the teplate to the coding standard.

**Comment by justinmclean** [comment]
Please refer to [1] which contains the correct text.

1. https://www.apache.org/legal/src-headers.html#headers

**Comment by davids5** [comment]
@Ouss4 - Thank you for the direct links! 

> We may want to add Github's templates for PRs/Issues. Maybe we bring this to the dev list for a discussion on what should be there?

@Ouss4 - email sent.

**Comment by davids5** [comment]
@justinmclean - Does you answer imply that the information given was incorrect or the contributor has it wrong or were you just giving an alternate source?

File: .gitignore [hunk 2]
```diff
@@ -29,4 +29,4 @@ core
 /.cproject
 /.project
 /tags
-/TAGS
+/TAGS
```
Comment (xiaoxiang781216) [inline .gitignore]: revert the change

Comment (aenrbes) [reply .gitignore]: I already fix it, and did a force push.

Comment (xiaoxiang781216) [reply .gitignore]: Just saw the new change, the PR looks fine now.

File: arch/risc-v/src/litex/hardware/litex_plic.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#ifndef __ARCH_RISCV_SRC_LITEX_HARDWARE_LITEX_PLIC_H
+#define __ARCH_RISCV_SRC_LITEX_HARDWARE_LITEX_PLIC_H
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* litex vexRiscv does not follow RISC-V privileged specification and
+ * uses two additional CSRs: mask and pending.
+ */
```
Comment (Ouss4) [inline arch/risc-v/src/litex/hardware/litex_plic.h]:
```suggestion
 */
 
```
Not sure why nxstyle didn't pick this one up.

File: arch/risc-v/src/litex/hardware/litex_plic.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
```
Comment (Ouss4) [inline arch/risc-v/src/litex/hardware/litex_plic.h]:
```suggestion

```
That must have been introduced by my Code suggestion. Sorry.

**Comment by Ouss4** [comment]
I think the PR looks good and is ready to merge. Let's give some time for anyone that wants to add any review. 
When merging, I'd like to squash the 5 commits into one.
I'll not bring the PR locally and do the squashing manually, I'll squash from Github.
